power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/dataset.hpp

```diff
@@ -88,8 +88,19 @@ class DatasetWritable {
 
 class DatasetMutable {
   public:
-    DatasetMutable(std::string const& dataset, Idx is_batch, Idx batch_size)
-        : dataset_{handle_.call_with(PGM_create_dataset_mutable, dataset.c_str(), is_batch, batch_size)},
+    template <typename T, typename = std::enable_if_t<std::is_same_v<T, Idx> || std::is_same_v<T, bool>>>
+    DatasetMutable(std::string const& dataset, T is_batch, Idx batch_size)
+        : handle_{},
+          dataset_{handle_.call_with(
+              PGM_create_dataset_mutable, dataset.c_str(),
+              [is_batch]() -> Idx {
+                  if constexpr (std::is_same_v<T, bool>) {
+                      return is_batch ? Idx{1} : Idx{0};
+                  } else {
+                      return is_batch;
+                  }
+              }(),
+              batch_size)},
```

Comment:
why such a complicated change?

Reply:
i wanted to get rid of using `0` and `1` instead of `false` and `true` for `is_batch`. Ran into the same bug multiple times in a row. I originally implemented it using `requires` but @figueroa1395 remembered that the C++ API should not use modern C++20 features but at the latest C++17.

Having 2 functions was not possible because MSVC thinks that `bool` and `Idx` are almost the same, so `enable_if` was not able to distinguish. This or some kind of external function are the only alternatives AFAICT

Reply:
I do not understand. We can just make `is_batch` as boolean input in the CPP wrapper right? It will be passed in to the C-API by upcasting it to `Idx`.

I see no need for such complicated stuff.